File: SecondYear/29deepanshutyagi_DeepanshuTyagi_2226cs1195_2/Week_2/factorial.cpp
#include <vector>
using namespace std;

vector<long long> factorialNumbers(long long n) {
    vector<long long> result;
    long long fact = 1;
    long long i = 1;
    
    while (fact <= n) {
        result.push_back(fact);
        i++;
        fact *= i;
    }
    
    return result;
}

File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/Week_2/Factorial.java
import java.util.ArrayList;
import java.util.List;

public class Factorial {

    static long fact(long j) {
        if (j == 1 || j == 0) {
            return 1;
        } else {
            return j * fact(j - 1);
        }
    }

    public static List<Long> factorialNumbers(long n) {
        List<Long> l = new ArrayList<>();
        for (long i = 1; i <= n; i++) {
            long a = fact(i);
            if (a <=n) {
                l.add(a);
            } else {
                break;
            }
        }
        return l;
    }
}
